File: code/sdk/android/cloudsheet-Android-sdk/src/main/kotlin/com/opencloudsheet/model/file/OneDriveFile.kt
package com.opencloudsheet.model.file

import com.opencloudsheet.response.onedrive.DriveItem

class OneDriveFile(
    private val item: DriveItem
) : ICloudFile {
    override fun getId(): String = item.id
    override fun getName(): String = item.name
    override fun isFolder(): Boolean = item.folder != null
    override fun getPath(): String? = item.parentReference?.path
    override fun getSize(): Long? = item.size?.takeIf { it >= 0 }
}

File: code/sdk/ios/OpenCloudSheet/Sources/OpenCloudSheet/Model/File/OneDriveFile.swift
//
//  OneDriveFile.swift
//  OpenCloudSheet
//

import Foundation

class OneDriveFile: ICloudFile {
    private let item: DriveItem

    init(item: DriveItem) {
        self.item = item
    }

    func getId() -> String {
        return item.id
    }

    func getName() -> String {
        return item.name
    }

    func isFolder() -> Bool {
        return item.folder != nil
    }

    func getPath() -> String? {
        return item.parentReference?.path
    }

    func getSize() -> Int64? {
        if let size = item.size, size >= 0 {
            return size
        }
        return nil
    }
}
